code/modules/projectiles/guns/flamer/flamer.dm

```diff
@@ -654,9 +661,19 @@ GLOBAL_LIST_EMPTY(flamer_particles)
 		thing.handle_flamer_fire(src, damage, delta_time)
 
 	//This has been made a simple loop, for the most part flamer_fire_act() just does return, but for specific items it'll cause other effects.
-	firelevel -= 2 //reduce the intensity by 2 per tick
+
+	firelevel -= 2 + weather_smothering_strength //reduce the intensity by 2 as default or more if in weather ---- weather_smothering_strength is set as /datum/weather_event's fire_smothering_strength
+
 	return
 
+/obj/flamer_fire/proc/update_in_weather_status()
+	SIGNAL_HANDLER
+	var/area/A = get_area(src)
+	if(SSweather.is_weather_event && locate(A) in SSweather.weather_areas)
```

Comment:
oogh really, that's not correct, `weather_areas` only contains master areas, not subareas

it probably work in practice because flames create light, causing everything to be shifted in the master area, but that's just not something you can count on

really, using `A.master in weather_areas` should solve that, look back at setup_weather_areas:
```
	for(var/area/A in all_areas)
		if(A == A.master && A.weather_enabled && map_holder.should_affect_area(A))
			weather_areas += A
```